sys/psa_crypto/psa_crypto.c

```diff
@@ -0,0 +1,1856 @@
+/*
+ * Copyright (C) 2021 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     sys_psa_crypto
+ * @{
+ *
+ * @file
+ * @brief       PSA Crypto API implementation
+ *
+ * @author      Lena Boeckmann <lena.boeckmann@haw-hamburg.de>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+#include "psa/crypto.h"
+#include "psa_crypto_se_driver.h"
+#include "psa_crypto_se_management.h"
+#include "psa_crypto_slot_management.h"
+#include "psa_crypto_location_dispatch.h"
+#include "psa_crypto_algorithm_dispatch.h"
+
+#include "random.h"
+#include "kernel_defines.h"
+
+#define ENABLE_DEBUG    0
+#include "debug.h"
+
+static uint8_t lib_initialized = 0;
+
+/**
+ * @brief   Compares the content of two same-sized buffers while maintaining
+ *          constant processing time
+ *
+ * @param   a Buffer A to compare with B
+ * @param   b Buffer B to compare with A
+ * @param   n Size of the input buffers
+ *
+ * @return  int
+ *          0 if buffer contents are the same
+ *          1 if buffer contents differ
+ */
+static inline int safer_memcmp(const uint8_t *a, const uint8_t *b, size_t n)
```

Comment:
Why not name it e.g. `constant_time_memcmp`? And if you insist on calling it `safer`, the comment should explain why constant time is relevant in a security context, timing attacks etc.

Reply:
reminds me of the one we have [in `shell_lock`](https://github.com/RIOT-OS/RIOT/blob/master/sys/shell_lock/shell_lock.c#L76)

Reply:
If one already exists would it make sense to implement a system wide constant time memcmp?

Reply:
Also done